ui/apps/platform/src/Containers/Search/SearchResults.tsx

```diff
@@ -223,20 +226,27 @@ function SearchResults({
         };
 
     const onFilterLinkHandler =
-        (searchCategory: string, category: string, toURL: string, name: string) => () => {
+        (searchCategory: string, category: string, toURL: string, name: string, location: string) =>
+        () => {
             const searchOptions = amendSearchOptions(searchCategory, name);
             passthroughGlobalSearchOptions(searchOptions, category);
             const searchFilter = searchOptionsToSearchFilter(searchOptions);
-            let queryString = '';
-            // TODO Once Violations, Risk, and Network Graph are all using URL Search Params this can be simplified
-            if (category === 'ALERTS' || category === 'DEPLOYMENTS') {
-                queryString = getUrlQueryStringForSearchFilter(searchFilter);
-            } else if (category === 'NETWORK') {
-                // Do nothing, until Network Graph moves from Redux to URL Params
+            const queryString = getUrlQueryStringForSearchFilter(searchFilter);
+            if (category === 'NETWORK') {
+                // Parse the cluster and namespace of the search result in order
```

Comment:
For better UX, we will want to set the selected cluster and namespace of the global search result when navigating to the Network Graph.

Although we have this information at this point in the code, there are a handful or problems with this:

1. The cluster and namespace _names_ are parsed from the `location` field returned by the search API. This code makes assumptions on the format of this string, which feels somewhat unreliable.
2. The cluster and ns URL hooks require _ids_ instead of _names_, which is why the `clusters` value from the Redux store needs to be pulled in here.
3. I'm not yet sure if there is a clean way to get a namespace id is here, given a cluster id and the _name_ of the namespace.

Ideally, we would update the search API to return the cluster and namespace ids, if available, in addition to the "location" field.